tasks/nc_apps.yml

```diff
@@ -2,59 +2,46 @@
   ansible.builtin.set_fact:
     nc_app_name: "{{ item.key }}"
+    nc_app_source: "{{ item.value.source if item.value.source is defined else item.value }}"
     nc_app_cfg: "{{ item.value }}"
 
-- name: "Install and enable Nextcloud Apps"
-  when: nc_app_cfg is string
+- name: "[App] - Download, install, enable and if provided configure Nextcloud Apps"
   block:
-    - name: "[App] - Download Archive in apps folder {{ nc_app_name }}"
-      ansible.builtin.unarchive:
-        copy: false
-        src: "{{ nc_app_cfg }}"
-        dest: "{{ nextcloud_webroot }}/apps/"
-        owner: "{{ nextcloud_websrv_user }}"
-        group: "{{ nextcloud_websrv_group }}"
-        creates: "{{ nextcloud_webroot }}/apps/{{ nc_app_name }}"
-      when: nc_app_cfg is not none
+    - name: "[App] - Verify the yaml declaration for the app \"{{ nc_app_name }}\""
+      ansible.builtin.assert:
+        that:
+          - (nc_app_cfg.source is defined) and (nc_app_cfg.source is string)
+        msg: "{{ nc_app_name }} is not well declared."
+      when: nc_app_cfg is not string
 
-    - name: "[App] - Enable the application {{ nc_app_name }}"
+    - name: "[App] - Install the app \"{{ nc_app_name }}\" from the official app-store"
       become_user: "{{ nextcloud_websrv_user }}"
       become_flags: "{{ ansible_become_flags | default(omit) }}"
       become: true
-      ansible.builtin.command: php occ app:enable "{{ nc_app_name }}"
+      ansible.builtin.command: php occ app:install "{{ nc_app_name }}"
       args:
         chdir: "{{ nextcloud_webroot }}"
-      when: nc_app_cfg is not none
-
-- name: "[App] - Install Apps"
-  when: nc_app_cfg is mapping
-  block:
-    - name: "[App] - Verify the app's yaml declaration"
-      ansible.builtin.assert:
-        that:
-          - (nc_app_cfg.source is defined) and (nc_app_cfg.source is string)
-        msg: "{{ nc_app_name }} is not well declared."
+      when: (nc_app_source is string) and (nc_app_source | length == 0)
 
-    - name: "[App] - Download Archive in apps folder {{ nc_app_name }}"
+    - name: "[App] - Download Archive in apps folder \"{{ nc_app_name }}\""
       ansible.builtin.unarchive:
         copy: false
-        src: "{{ nc_app_cfg.source }}"
+        src: "{{ nc_app_source }}"
         dest: "{{ nextcloud_webroot }}/apps/"
         owner: "{{ nextcloud_websrv_user }}"
         group: "{{ nextcloud_websrv_group }}"
         creates: "{{ nextcloud_webroot }}/apps/{{ nc_app_name }}"
-      when: nc_app_cfg.source is not none
+      when: (nc_app_source is string) and (nc_app_source | length > 0)
 
-    - name: "[App] - Enable the application {{ nc_app_name }}"
+    - name: "[App] - Enable the application \"{{ nc_app_name }}\""
       become_user: "{{ nextcloud_websrv_user }}"
       become_flags: "{{ ansible_become_flags | default(omit) }}"
       become: true
       ansible.builtin.command: php occ app:enable "{{ nc_app_name }}"
       args:
         chdir: "{{ nextcloud_webroot }}"
```

Comment:
Missing `when:` clause

Reply:
I did test this part of the script in all variants and it worked properly. So not sure which "when" is missing here. The way nc_apps works now is different to the former version.

Reply:
I meant `when` for task in line L36:
`- name: "[App] - Enable the application \"{{ nc_app_name }}\""`
it will be run always, even when `nc_app_name` is empty or none.

Reply:
Can this happen? This part is called from loop in main.yaml.

Reply:
Hmm... We will see with molecule run.

Reply:
OK. I don't think so, because this is the loop-variable. Both cases shall not lead to the loop being run.